### ISSUE TITLE
Problems with IM Model

### ISSUE THREAD

**Issue by jeromekelleher** [popsim-consortium/stdpopsim#533]
The [IM model](https://stdpopsim.readthedocs.io/en/latest/api.html#stdpopsim.IsolationWithMigration) has a few issues:

1. Currently the parameters are ``stdpopsim.IsolationWithMigration(NA, N1, N2, T, M12, M21)``, but the actual populations are 0 == pop1, 1 == pop2, 2 = ancestral pop. It would be less confusing if we had `stdpopsim.IsolationWithMigration(N0, N1, NA, T, M01, M10)`, and the populations then at least matched their indexes in the final tree sequence.
2. The population indexes aren't documented in the output
3. What direction of time is the migration rate M01/M10 going in?
4. There's a redundant sentence at the start of the docstring (same for PiecewiseConstantSize).

In general, I think these models would be better off defined in msprime -- see https://github.com/tskit-dev/msprime/pull/1025/ for a proposal.

It's probably still good to have an interface to them here in stdpopsim though.

**Comment by andrewkern** [comment]
so pull the generic models out of stdpopsim completely?

**Comment by jeromekelleher** [comment]
No, I'd say we keep them as options in stdpopsim, but these would be basically calls to msprime functions.

**Comment by apragsdale** [comment]
I think I'm to blame for the awkward implementation of the IM model. Sorry about that. But I do think it's important to keep the generic models in stdpopsim. I've already been using the ability to easily couple the multiple epoch or IM models with the contigs/maps/etc.

**Comment by andrewkern** [comment]
> But I do think it's important to keep the generic models in stdpopsim. 

+1. a couple of people have asked about this already

**Comment by jeromekelleher** [comment]
Sure, let's keep them here. I just think it makes more sense to *define* them in msprime, as they are quite different to the bread-and-butter stdpopsim stuff.

**Comment by apragsdale** [comment]
Let's set this for the next release (0.1.3 instead of 0.1.2), since we're waiting for the generic models in msprime? When generic models on the msprime end get set up and released, I'm happy to refactor this then and perhaps add other relevant generic models at the same time.

**Comment by jeromekelleher** [comment]
OK, sounds good. We'll have to document potential breakage, but since it's not a major feature of stdpopsim at the moment we don't have to worry too much about that.

**Comment by grahamgower** [comment]
Msprime now has some an island_model() function (and other generics).
https://tskit.dev/msprime/docs/stable/api.html#msprime.Demography.isolated_model
https://tskit.dev/msprime/docs/stable/api.html#msprime.Demography.island_model
https://tskit.dev/msprime/docs/stable/api.html#msprime.Demography.stepping_stone_model

**Comment by nspope** [comment]
It'd be good to use a consistent naming scheme for populations in the generic models-- eg use `pop_0`, `pop_1` rather than `pop1`, `pop2` for `IsolationWithMigration`, to match `PiecewiseConstantSize`.

**Comment by petrelharp** [comment]
The first step here is to figure out whether the associated changing of e.g. population names would cause too much breakage for others?

**Comment by mufernando** [comment]
I only found one piece of code on Github using the IsolationWithMigration model (pg-gan), but the PiecewiseConstantSize is being used by many different projects (e.g., phlash, mush, our own analysis2).

In a call with @andrewkern @petrelharp, we decided that we should leave this for the next release. Idea is to create new functions (new names) and add deprecation warning to these old functions.
